src/litserve/server.py

```diff
@@ -220,21 +231,67 @@ def launch_inference_worker(self, num_uvicorn_servers: int):
                 raise e
 
         process_list = []
-        for worker_id, device in enumerate(self.devices * self.workers_per_device):
+        # for worker_id, device in enumerate(self.devices * self.workers_per_device):
+        #     if len(device) == 1:
+        #         device = device[0]
+
+        #     self.workers_setup_status[worker_id] = False
+
+        #     ctx = mp.get_context("spawn")
+        #     process = ctx.Process(
+        #         target=inference_worker,
+        #         args=(
+        #             self.lit_api,
+        #             self.lit_spec,
+        #             device,
+        #             worker_id,
+        #             self.request_queue,
+        #             self.response_queues,
+        #             self.max_batch_size,
+        #             self.batch_timeout,
+        #             self.stream,
+        #             self.workers_setup_status,
+        #         ),
+        #     )
+        #     process.start()
+        #     process_list.append(process)
+        for worker_id, device in enumerate(self.preprocess_workers):
             if len(device) == 1:
                 device = device[0]
-
             self.workers_setup_status[worker_id] = False
```

Comment:
This is incorrect, because `worker_id` on line 285 will overwrite the same entries in the dictionary.
I mentioned we can remove the setup status for preprocess, because there's no setup in the first place, so we can just remove this line.